src/net.cpp

```diff
@@ -1650,14 +1664,12 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
                 // We will not make outgoing connections to peers that are unreachable
                 // (e.g. because of -onlynet configuration).
                 // Therefore, we do not add them to addrman in the first place.
-                // Note that if you change -onlynet setting from one network to another,
-                // peers.dat will contain only peers of unreachable networks and
-                // manual intervention will be needed (either delete peers.dat after
-                // configuration change or manually add some reachable peer using addnode),
-                // see <https://github.com/bitcoin/bitcoin/issues/26035> for details.
+                // In case previously unreachable networks become reachable
+                // (e.g. in case of -onlynet changes by the user), fixed seeds will
+                // be loaded only for networks for which no we have no addressses.
                 seed_addrs.erase(std::remove_if(seed_addrs.begin(), seed_addrs.end(),
-                                               [](const CAddress& addr) { return !IsReachable(addr); }),
-                                seed_addrs.end());
+                                                [fixed_seed_networks](const CAddress& addr) { return std::find(fixed_seed_networks.begin(), fixed_seed_networks.end(), addr.GetNetwork()) == fixed_seed_networks.end(); }),
```

Comment:
A vector is usually not cheap to copy. Better capture it by reference:
```suggestion
                                                [&fixed_seed_networks](const CAddress& addr) { return std::find(fixed_seed_networks.begin(), fixed_seed_networks.end(), addr.GetNetwork()) == fixed_seed_networks.end(); }),
```